sdks/java/io/synthetic/src/test/java/org/apache/beam/sdk/io/synthetic/CoGroupByKeyLoadIT.java

```diff
@@ -0,0 +1,140 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.synthetic;
+
+import static org.apache.beam.sdk.io.synthetic.SyntheticSourceTestUtils.fromString;
+
+import java.io.IOException;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.PipelineOptionsValidator;
+import org.apache.beam.sdk.options.Validation;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.testing.TestPipelineOptions;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.join.CoGbkResult;
+import org.apache.beam.sdk.transforms.join.CoGroupByKey;
+import org.apache.beam.sdk.transforms.join.KeyedPCollectionTuple;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TupleTag;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Performance fanout test for {@link CoGroupByKey}. */
+@RunWith(JUnit4.class)
+public class CoGroupByKeyLoadIT {
+
+  private static Options options;
+
+  private static SyntheticBoundedIO.SyntheticSourceOptions inputOptions;
+
+  private static SyntheticBoundedIO.SyntheticSourceOptions coInputOptions;
+
+  private static final TupleTag<byte[]> INPUT_TAG = new TupleTag<>();
+  private static final TupleTag<byte[]> CO_INPUT_TAG = new TupleTag<>();
+
+  @Rule public TestPipeline pipeline = TestPipeline.create();
+
+  /** Pipeline options for the test. */
+  public interface Options extends TestPipelineOptions {
+
+    @Description("The JSON representation of SyntheticBoundedInput.SourceOptions. Input options.")
+    @Validation.Required
+    String getInputOptions();
+
+    void setInputOptions(String  inputOptions);
+
+    @Description("The JSON representation of SyntheticBoundedInput.SourceOptions. Co-input options.")
+    @Validation.Required
+    String getCoInputOptions();
+
+    void setCoInputOptions(String inputOptions);
+
+    @Description("The number of gbk to perform")
+    @Default.Integer(5)
+    Integer getFanout();
+
+    void setFanout(Integer shuffleFanout);
+  }
+
+  @BeforeClass
+  public static void setup() throws IOException {
+    PipelineOptionsFactory.register(Options.class);
+
+    options =
+        PipelineOptionsValidator.validate(
+            Options.class, TestPipeline.testingPipelineOptions().as(Options.class));
+
+    inputOptions = fromString(options.getInputOptions());
+    coInputOptions = fromString(options.getCoInputOptions());
+  }
+
+  // TODO: Is this a proper fanout test?
+  // TODO: Should this be a fanout test too or should we just "reiterate large inputs"?
```

Comment:
This file looks good to me. I have two questions that I can't answer:
- Do we want to make this a fanout test?
- How do we make sure that the sources generate data that will group together? (i.e. with matching keys)

We can talk with Marian about this.